lib/x86simdsort.h

```diff
@@ -3,11 +3,28 @@
 #include <stdint.h>
 #include <vector>
 #include <cstddef>
+#include <functional>
 
 #define XSS_EXPORT_SYMBOL __attribute__((visibility("default")))
 #define XSS_HIDE_SYMBOL __attribute__((visibility("hidden")))
 #define UNUSED(x) (void)(x)
 
+template <typename T>
+XSS_HIDE_SYMBOL void permute_array_in_place(T *A, std::vector<size_t> P)
```

Comment:
Move to an `x86simdsort::detail`  namespace.

Take the `P` parameter by const-reference or change the call site to use `std::move(arg)`.